repos/spack_repo/builtin/packages/pbvr/package.py

```diff
@@ -0,0 +1,148 @@
+import re
+
+from spack_repo.builtin.build_systems.makefile import MakefilePackage
+
+from llnl.util.filesystem import install_tree
+
+from spack.package import *
+from spack.util.environment import set_env
+
+
+class Pbvr(MakefilePackage):
+    """CS/IS-PBVR is a scientific visualization application designed
+    based on Particle-Based Volume Rendering (PBVR). This application
+    is capable of multivariate visualization and three-dimensional point cloud visualization
+    in addition to standard visualization functions such as volume rendering
+    and isosurfaces for 3D volume data obtained from simulations and measuring instruments.
+    In addition, the framework for distributed processing of optimized PBVR is characterized
+    by the ability to remotely visualize large-scale time-series volume data in remote locations
+    at high speed. As a method of remote visualization, you can choose between
+    client-server (CS) visualization, which visualizes volume data stored in remote storage,
+    and in-situ (IS) visualization, which visualizes simulations simultaneously and
+    in the same environment. This application is being developed at the Center
+    for Computational Science and e-Systems of the Japan Atomic Energy Agency."""
+
+    phases = ["build", "install"]
+
+    homepage = "https://github.com/CCSEPBVR/CS-IS-PBVR"
+    url = "https://github.com/CCSEPBVR/CS-IS-PBVR/archive/refs/tags/v3.5.0.tar.gz"
+
+    maintainers("sakamoto-naohito")
+
+    license("LGPL-3.0-only")
+
+    version("3.5.0", sha256="264c82d9e94b6f8477952ce2f80834332dbc9047db694f7f3ba2ab07c7c92aae")
+
+    variant("client", default=True, description="Build Client Program")
+    variant("mpi", default=True, description="Enable MPI Support")
+    variant("extended_fileformat", default=True, description="Enable extended fileformat")
+
+    depends_on("gmake", type="build")
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+
+    depends_on("mpi", when="+mpi")
+    depends_on("qt-base@6.2.4+opengl", when="+client")
+    depends_on("qt-svg@6.2.4+widgets", when="+client")
+    depends_on("vtk@9.3.1~mpi", when="~mpi")
+    depends_on("vtk@9.3.1+mpi", when="+mpi")
+    depends_on("freeglut", when="+client")
+
+    patch("kvs-conf.patch", when="~client~extended_fileformat")
+    patch("kvs-extended-fileformat-conf.patch", when="~client+extended_fileformat")
+    patch("kvs-client-conf.patch", when="+client~extended_fileformat")
+    patch("kvs-client-extended-fileformat-conf.patch", when="+client+extended_fileformat")
+
+    patch("pbvr-conf.patch", when="~mpi")
+    patch("pbvr-conf-mpi.patch", when="+mpi")
+    patch("makefile-machine-gcc-omp.patch", when="~mpi")
+    patch("makefile-machine-gcc-mpi-omp.patch", when="+mpi")
+
+    def setup_build_environment(self, env):
+        compiler_name = self.spec.compiler.name
+        if compiler_name != "gcc":
+            raise InstallError(f"{self.name} is only supported with the GCC compiler.")
```

Comment:
This is better handled as a package requirement so the error is emitted on concretization, not building.
```suggestion
    requires("%cxx=gcc")
```